lib/api/apiUtils/bucket/bucketCors.js

```diff
@@ -0,0 +1,278 @@
+            return undefined;
+        }
+        return errors.MalformedXML
+            .customizeDescription(errs.originAndMethodExist);
+    },
+    validateMethods(methods) {
+        let invalidMethod;
+        function isValidMethod(method) {
+            if (method !== 'GET' && method !== 'PUT' && method !== 'HEAD' &&
+            method !== 'POST' && method !== 'DELETE') {
+                invalidMethod = method;
+                return false;
+            }
+            return true;
+        }
+        if (!methods.every(isValidMethod)) {
+            const errMsg = 'Found unsupported HTTP method in CORS config. ' +
+            `Unsupported method is "${invalidMethod}"`;
+            return errors.InvalidRequest.customizeDescription(errMsg);
+        }
+        return undefined;
+    },
+    validateOrigins(origins) {
+        for (let i = 0; i < origins.length; i++) {
+            const origin = origins[i];
+            const errMsg = `AllowedOrigin "${origin}" can not have ` +
+            'more than one wildcard.';
+            if (typeof origin !== 'string' || origin === '') {
+                return errors.MalformedXML;
+            }
+            const numberWildcards = (origin.match(/\*/g) || []).length;
+            if (numberWildcards > 1) {
+                return errors.InvalidRequest.customizeDescription(errMsg);
+            }
+        }
+        return undefined;
+    },
+    validateAllowedHeaders(headers) {
+        if (!headers) {
+            return undefined;
+        }
+        if (!Array.isArray(headers) || headers.length === 0) {
+            return errors.MalformedXML;
+        }
+        for (let i = 0; i < headers.length; i++) {
+            const header = headers[i];
+            const errMsg = `AllowedHeader "${header}" can not have ` +
+            'more than one wildcard.';
+            if (typeof header !== 'string' || header === '') {
+                return errors.MalformedXML;
+            }
+            // TODO: Consider adding validation for invalid symbols in header,
+            // even if AWS does not
+            const numberWildcards = (header.match(/\*/g) || []).length;
+            if (numberWildcards > 1) {
+                return errors.InvalidRequest.customizeDescription(errMsg);
+            }
+        }
+        return true; // to indicate AllowedHeaders exists
+    },
+    validateExposeHeaders(headers) {
+        if (!headers) {
+            return undefined;
+        }
+        if (!Array.isArray(headers) || headers.length === 0) {
+            return errors.MalformedXML;
+        }
+        for (let i = 0; i < headers.length; i++) {
+            const header = headers[i];
+            if (typeof header !== 'string') {
+                return errors.MalformedXML;
+            }
+            if (header.indexOf('*') !== -1) {
+                const errMsg = `ExposeHeader ${header} contains a wildcard. ` +
+                'Wildcards are currently not supported for ExposeHeader.';
```

Comment:
I copied the error message from AWS, so maybe they are thinking one day it will be supported.
Should I remove the 'currently' to reduce confusion?